### ISSUE TITLE
WCAG 2.2: Confirm an email address ‖ Accessible Authentication

### ISSUE THREAD

**Issue by querkmachine** [alphagov/govuk-design-system#2872]
## What

Update our guidance on confirming an email address to emphasise that developers should allow users to copy and paste verification codes into an input field, if verification codes are being used. 

### Resources

- [Confirm an email address guidance page](https://design-system.service.gov.uk/patterns/confirm-an-email-address/)
- [WCAG 2.2 criterion for 'Accessible Authentication (Minimum)'](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-minimum)

## Why

The new WCAG 2.2 Level AA criterion 'Accessible Authentication (Minimum)' says: 

> A [cognitive function test](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-minimum#dfn-cognitive-function-test) (such as remembering a password or solving a puzzle) is not required for any step in an authentication [process](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-minimum#dfn-process) unless that step provides at least one of the following: 
> 
> [...] 
> 
> **Mechanism**
>   A [mechanism](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-minimum#dfn-mechanism) is available to assist the user in completing the cognitive function test.
> 
> [...]
> 
> Examples of mechanisms that satisfy this criterion include:
> 1. support for password entry by password managers to reduce memory need, and
> 2. copy and paste to reduce the cognitive burden of re-typing.

Processes that require transcribing authentication codes or passwords, such as logging in or authenticating one's identity, are thus within scope of this criterion.

Doing this also satisfies the Level AAA criterion, ['Accessible Authentication (Enhanced)'](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-enhanced.html).

## Who needs to work on this
- Content designer or technical writer

## Who needs to review this
- Accessibility specialist
- Content designer or technical writer

## Done when

- [ ] Guidance has been updated

**Comment by dav-idc** [comment]
This has been added to the ['WCAG 2.2 content updates' document](https://docs.google.com/document/d/1nIxIKgVUheC8zvVdqWIcqtVPCl7oTN4hv11R9rDwdvM/edit?usp=sharing).

**Comment by dav-idc** [comment]
This issue appears to already be covered by the existing planned guidance updates in the WCAG 2.2 content updates main guidance doc.
Here's the existing section that covers footers and consistent help: https://docs.google.com/document/d/1nIxIKgVUheC8zvVdqWIcqtVPCl7oTN4hv11R9rDwdvM/edit#heading=h.varqk0aj5n9z

As a result I'm closing this issue as a duplicate of the main WCAG 2.2 guidance issue:
- https://github.com/alphagov/govuk-design-system/issues/2729
- https://github.com/alphagov/govuk-design-system/pull/3090

**Comment by calvin-lau-sig7** [comment]
Closing issue – content has been drafted, reviewed and added to pull request #3090 as part of https://github.com/alphagov/govuk-design-system/issues/3276.

**Comment by mtcA6** [comment]
What isn't clear in the guidance is how long any verification process should be valid. In other words how long should a link or code or other verification tool last before it expires. Is there a minimum?
